lib/Horde/String.php

```diff
@@ -199,7 +199,11 @@ public static function lower($string, $locale = false, $charset = null)
         if (!isset(self::$_lowers[$string])) {
             $language = setlocale(LC_CTYPE, 0);
             setlocale(LC_CTYPE, 'C');
-            self::$_lowers[$string] = strtolower($string);
+            if ($string === null) {
+                self::$_lowers[$string] = '';
+            } else {
+                self::$_lowers[$string] = strtolower($string);
```

Comment:
Isn't it simply?
```suggestion
                self::$_lowers[$string] = strtolower($string ?? '');
```

Reply:
Yes except `??` is PHP 7.0. We're using ternary instead on my fork, @ChristophWurst just haven't updated this PR to match.